docs/zkapps/advanced-snarkyjs/actions-and-reducer.mdx

```diff
@@ -99,9 +99,63 @@ this.reducer.getActions({ fromActionsHash?: Field, endActionsHash?: Field }): A[
 
 Right now, `getActions` is available for [testing with `LocalBlockchain`](/zkapps/how-to-test-a-zkapp).
 
+### Fetching Actions from an Archive Node
+
+Since Mina nodes do not store historical network information and actions are not kept on-chain, emitted actions are only preserved in the Archive Node. Consequently, a zkApp must fetch its previously emitted actions on the Mina network from an Archive Node.
+
+As the Archive Node is essentially a database, it does not offer a user-friendly API that SnarkyJS can readily utilize to fetch actions. To address this issue, a GraphQL server was developed to retrieve network information such as actions and actions from the Archive Node. SnarkyJS will use this GraphQL server to fetch actions from the Archive Node.
+
+To configure SnarkyJS to use an Archive Node to fetch actions, use the `archive` property in the `Mina.Network` function. The `archive` property must be specified in the `Mina.Network` function to fetch actions from an Archive Node; otherwise, an error will occur.
+
+You can set the `archive` property to one of the following endpoints:
+
+```sh
+https://api.minascan.io/archive/berkeley/v1/graphql
+```
+
+```sh
+https://archive.berkeley.minaexplorer.com
+```
+
+For example, to use the `archive` property, see the following example code:
+
+```ts
+const Network = Mina.Network({
+  mina: '', // Use https://proxy.berkeley.minaexplorer.com/graphql or https://api.minascan.io/node/berkeley/v1/graphql
+  archive: '', // Use https://api.minascan.io/archive/berkeley/v1/graphql/ or https://archive.berkeley.minaexplorer.com/
+});
+Mina.setActiveInstance(Network);
+```
+
+By setting the `archive` property to one of the listed endpoints, SnarkyJS uses the selected endpoint to fetch events. For instance, if a zkApp has already emitted events and broadcasted the transaction, you can use the `getActions()` function to retrieve actions for a zkApp:
+
+```ts
+class MyContract extends SmartContract {
+  ...
+  @method getActionsExample() {
+    // Get all actions for this zkApp
+    let pendingActions = this.reducer.getActions({
+      fromActionHash: actionsHash,
+    });
+    this.reducer.reduce(
+      pendingActions,
+      Field,
+      (state: Field, action: Field) => {
+        ...
+      },
+      { state: counter, actionsHash }
+    );
+  }
+}
+```
+
+The `getActions()` function accepts an optional `fromActionHash` parameter. This `actionsHash` is derived from processing Account Updates and indicates the starting point for your zkApp to begin processing actions. If `fromActionHash` is not provided, `getActions()` get all actions for the zkApp, since the first one. The function returns a list of actions emitted after the specified `fromActionHash` has been processed.
```

Comment:
```suggestion
By default, `getActions()` retrieves all actions from the very first ever emitted for that zkApp account. However, you can provide an object containing an optional property named `fromActionHash` where the value is an `actionsHash` to indicate the starting point of the actions to be retrieved and processed by your smart contract. Note that actions _after_ the specified hash will be retrieved—i.e. it is not inclusive.
```

@MartinMinkov From your text, it appears that the specified actionHash is excluded from the returned results. Is that correct? That's somewhat surprising behavior, no? Is there is a reason it can't or shouldn't be inclusive?

Reply:
It seems awkward at first, but I still promote present tense, even for future actions
In this case
will be retrieved >> is retrieved

https://developers.google.com/style/tense